src/Bicep.Core/TypeSystem/TypeAssignmentVisitor.cs

```diff
@@ -84,19 +85,17 @@ public IEnumerable<IDiagnostic> GetAllDiagnostics()
 
         private void AssignTypeWithCaching(SyntaxBase syntax, Func<TypeAssignment> assignFunc)
         {
+            // we could have used GetOrAdd here, but we don't need the return value
```

Comment:
Why not just use it anyway? Feels like it would simplify the code, and makes it a little harder to accidentally introduce race conditions in the future.

Reply:
Yeah good pt. I did something similar in the `DeclaredTypeManager`.